fmriprep/interfaces/images.py

```diff
@@ -355,70 +392,65 @@ def _run_interface(self, runtime):
             open(out_report, 'w').close()
             self._results['out_report'] = out_report
             return runtime
 
+        # A new file will be written
         out_fname = fname_presuffix(self.inputs.in_file, suffix='_valid', newpath=runtime.cwd)
         self._results['out_file'] = out_fname
 
-        warning = False
-        warning_txt = ''
-        description = ''
-        # Fix sform if needed
-        if not valid_codes:
-            # Nibabel derives a default LAS affine from the shape and zooms
-            # Use scanner xform code to indicate no alignment has been done
-            img.set_sform(img.affine, nb.nifti1.xform_codes['scanner'])
-            warning = True
-            warning_txt = 'Invalid header'
-            description += """\
-<p class="elem-desc">Input file does not have valid qform or sform matrix.
-  A default, LAS-oriented affine has been constructed.
-  A left-right flip may have occurred.
-  Analyses of this dataset MAY BE INVALID.
+        # Row 2:
+        if valid_qform and qform_code > 0 and sform_code == 0:
+            img.set_sform(img.get_qform(), qform_code)
+            warning_txt = 'Resetted sform matrix'
+            description = """\
+<p class="elem-desc">
+    The code of the sform matrix is "unknown", but code and matrix of qform are valid.
+    The sform- matrix and code have been replaced with the qform- matrix and code.
 </p>
 """
-        if not valid_qform and old_qform_code != 0:
-            if not warning:
-                warning = True
-                warning_txt = 'Invalid qform'
-            description += """\
-<p class="elem-desc">Input file does not have a valid qform matrix.</p>
+        # Rows 3-4:
+        elif (not matching_affines and sform_code > 0) or (qform_code == 0 and sform_code > 0):
+            img.set_qform(img.get_sform(), sform_code)
+            warning_txt = 'Resetted qform matrix'
+            description = """\
+<p class="elem-desc">
+    The sform matrix and code were valid, but the qform matrix did not match the sform matrix
+    or its code was "unknown". The qform- matrix and code have been replaced with the
+    sform- matrix and code.
+</p>
 """
-
-        if not matching_affines:  # Guaranteed if not valid_qform
-            if not warning:
-                warning_txt = 'Mismatched qform/sform matrices'
-            description += '<p class="elem-desc">qform matrix synced to sform.</p>\n'
-            sform, sform_code = img.get_sform(coded=True)
-            # Copy sform into qform
-            img.set_qform(sform, int(sform_code))
-
-        snippet = '<h3 class="elem-title">%s%s</h3>\n%s\n' % ('WARNING - ' if warning else '',
-                                                              warning_txt, description)
-
+        # Rows 5-6:
+        else:
+            img.set_sform(img.affine, nb.nifti1.xform_codes['scanner'])
+            img.set_qform(img.affine, nb.nifti1.xform_codes['scanner'])
```

Comment:
Because the `set_?form` functions change the affine data, I think I'd prefer that we do:

```Python
affine = img.affine
img.set_sform(affine, ...)
img.set_qform(affine, ...)
```

It should be equivalent, but its just fewer potentially moving parts.